src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -378,6 +383,14 @@ protected override bool Read(ReadableBuffer readableBuffer, WritableBuffer writa
 
                 return _inputLength == 0;
             }
+
+            protected override void OnReadStart()
+            {
+                if (_contentLength > _context.MaxRequestBodySize)
+                {
+                    _context.RejectRequest(RequestRejectionReason.RequestBodyTooLarge);
```

Comment:
Renaming to `ThrowXxx` probably makes sense to indicate control is not returned to caller.